### PR TITLE
Initial Setup of HA API

### PR DESCRIPTION
**Pull request by kylepotts**

### CODE REVIEW

File: src/lib.rs
Comment (BCNelson) [inline src/lib.rs:65]: Removed Functionality

File: src/lib.rs
Comment (BCNelson) [inline src/lib.rs:330]: Removed Functionality

File: src/lib.rs
Comment (BCNelson) [inline src/lib.rs:276]: Removed Functionality

File: src/lib.rs
Comment (BCNelson) [inline src/lib.rs:207]: Not saving returned information

File: src/lib.rs
Comment (BCNelson) [inline src/lib.rs]: Hard coded Client Id

File: src/lib.rs
Comment (BCNelson) [inline src/lib.rs:265]: Removed Functionality

File: src/lib.rs
Comment (BCNelson) [inline src/lib.rs:123]: Removed Functionality

File: src/lib.rs
Comment (BCNelson) [inline src/lib.rs:123]: Removed Functionality

**Review by BCNelson — CHANGES_REQUESTED** [review]

File: src/errors.rs
Comment (RealOrangeOne) [inline src/errors.rs]: Maybe worth just using wildcard match for all these `None` cases?
```suggestion
            _=> None,
```

File: src/lib.rs
Comment (RealOrangeOne) [inline src/lib.rs]: What's the thought behind multiple instance URLs? For local/remote? Looks like most of the below code just does `[0]`

**Review by RealOrangeOne — COMMENTED** [review]
Overall this looks great :+1: Although I recommend waiting for https://github.com/selfhostedshow/ha_api/pull/2 to merge and then updating before merging this one.

File: src/lib.rs
Comment (kylepotts) [reply src/lib.rs]: This is more based off of @BCNelson work where he had multiple instances. I'm open to just using one.

File: src/lib.rs
Comment (kylepotts) [reply src/lib.rs:207]: Not quite sure I understand. What should we be saving here?

**Review by RealOrangeOne — APPROVED** [review]

File: src/lib.rs
Comment (BCNelson) [reply src/lib.rs:207]: we should not not requiring that the user wire together the webhook id etc.

Comment (kylepotts) [reply src/lib.rs:207]: Sorry I still don't think I understand what you are asking. What do you mean wire together the webhook id? AFAIK the webhook is will always come back as part of this response.

Comment (BCNelson) [reply src/lib.rs:207]: we should be saving the webhook id and not making the user do that. I'm fine with returning it so that they can access it but we should save it in the object.

Comment (BCNelson) [reply src/lib.rs:207]: Actually never mind its goo that should be part of a bigger change to the API. That we can work on later.

File: Cargo.toml
Comment (BCNelson) [inline Cargo.toml]: The problem with this is that it "Pollutes" the versions available on crates.io. I have no problem putting beta version on there but it means that updates here will need to be approved before updates to halcyon can be reviewed.

Comment (BCNelson) [reply Cargo.toml]: Also, this particular scheme is not supported by cargo [link to docs](https://doc.rust-lang.org/cargo/reference/resolver.html#version-metadata)

**Comment by BCNelson** [comment]
@kylepotts We are good to go if you change the version to 0.2.0

**Review by BCNelson — APPROVED** [review]

**Pull request merged**
